src/main/java/org/opentest4j/FileInfo.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.opentest4j;
+
+import java.io.Serializable;
+
+/**
+ * {@code FileInfo} is expected to be used as {@code actual} and {@code expected} values in {@code AssertionFailedError}
+ *
+ * @author Marc Philipp
+ * @author Reinhold Degenfellner
+ * @since 1.2.1
+ */
+public class FileInfo implements Serializable {
+
+	private static final long serialVersionUID = 1L;
+
+	private final String contents;
+	private final String path;
+
+	/**
+	 * Constructs a {@code FileInfo} with a contents and a path.
+	 *
+	 * @param contents the contents of the file; must not be {@code null}
+	 * @param path the path of the file; must not be {@code null}
+	 */
+	public FileInfo(String contents, String path) {
+		this.contents = contents;
+		this.path = path;
```

Comment:
Shouldn't this validate the non-null requirement? Otherwise it will fail in `equals`/`hashCode`.